### PR TITLE
Enable publishing in VMR

### PR DESCRIPTION
**Pull request by NikolaMilosavljevic**
Contributes to https://github.com/dotnet/source-build/issues/4101

Enables publishing in VMR.

This repo creates packages in locations that aren't collected by common publishing infra in arcade. We need to collect them here.

### CODE REVIEW

**Comment by NikolaMilosavljevic** [comment]
Will update this to match similar changes in symreader and fsharp.

**Review by ViktorHofer — COMMENTED** [review]
Why is this PR needed? Isn't that the default? Which packages aren't picked up?

**Comment by mmitche** [comment]
Yeah, agreed that these should already be the defaults

**Comment by mmitche** [comment]
I think there is some subtlety here. The release package generation (generates release and prerelease versions of the shipping packages) sticks everything under packages\Release and packages\PreRelease. All of it needs to be signed, but only the shipping and nonshipping dirs are reported to BAR for dependency flow.

**Comment by ViktorHofer** [comment]
@NikolaMilosavljevic which packages are currently not picked up by publishing? Please give an example.

**Comment by NikolaMilosavljevic** [comment]
> @NikolaMilosavljevic which packages are currently not picked up by publishing? Please give an example.

With changes in this PR, all SBE packages are picked up. Without the changes, none of the packages are picked up. Just 2 examples:
```
Humanizer.Core.2.14.1.nupkg
Microsoft.ApplicationInsights.2.21.0.nupkg
```

This is due to different sub-folder naming in packages folder. Default arcade infra collects packages from `Shipping` and `NonShipping` folders. SBE packages are in `Release` sub-folder.

**Comment by ViktorHofer** [comment]
> SBE packages are in Release sub-folder.

Interesting. Why is that so?

**Review by ViktorHofer — APPROVED** [review]
Approving to get things flowing but we should follow-up on why SBE puts the artifacts under packages/Release instead of the expected directories.

File: eng/Publishing.props [hunk 1]
```diff
@@ -1,6 +1,10 @@
-<?xml version="1.0" encoding="utf-8"?>
 <Project>
-  <PropertyGroup>
-    <PublishingVersion>3</PublishingVersion>
-  </PropertyGroup>
+
+  <ItemGroup>
+    <ItemsToPushToBlobFeed Include="$(ArtifactsPackagesDir)**\*.nupkg"
+                           IsShipping="true"
+                           UploadPathSegment="Runtime"
```
Comment (ViktorHofer) [inline eng/Publishing.props:6]: @NikolaMilosavljevic I don't think this line is necessary and should probably be deleted. Please also add a tracking issue for investigating why packages are not in the expected subfolders.

Comment (NikolaMilosavljevic) [reply eng/Publishing.props:6]: Created an issue to fix this and remove custom publishing infra: https://github.com/dotnet/source-build/issues/4235

**Pull request merged**